datashuttle/command_line_interface.py

```diff
@@ -0,0 +1,662 @@
+    """
+    If project is protected test name, dump the variables to
+    stdout so they can be checked. Otherwise, run the
+    DataShuttle function.
+    """
+    if project.project_name == PROTECTED_TEST_PROJECT_NAME:
+        print("TEST_OUT_START: ", simplejson.dumps([args, kwargs]))
+    else:
+        function(*args, **kwargs)
+
+
+def make_kwargs(args):
+    kwargs = vars(args)
+    del kwargs["func"]
+    del kwargs["project_name"]
+    return kwargs
+
+
+def handle_bool(key, value):
+    if key in [
+        "ssh_to_remote",
+        "use_ephys",
+        "use_ephys_behav",
+        "use_ephys_behav_camera",
+        "use_behav",
+        "use_behav_camera",
+        "use_imaging",
+        "use_histology",
+    ]:
+        if value is not None and value not in [
+            "True",
+            "False",
+            "true",
+            "false",
+        ]:
+            utils.raise_error("Input value must be True or False")
+
+        value = value in ["True", "true"]
+
+    elif value in ["None", "none"]:
+        value = None
+
+    return value
+
+
+def handle_kwargs_bools(kwargs):
+    """"""
+    for key in kwargs.keys():
+        kwargs[key] = handle_bool(key, kwargs[key])
+    return kwargs
+
+
+def help(help_type):
+    """ """
+
+    if help_type == "flag_default_false":
+        help_str = "flag (default False)"
+
+    elif help_type == "required_str":
+        help_str = "Required: (str)"
+
+    elif help_type == "optional_flag_default_false":
+        help_str = "Optional: flag (default False)"
+
+    elif help_type == "required_str_single_or_multiple":
+        help_str = "Required: (str, single or multiple)"
+
+    elif help_type == "required_str_single_or_multiple_or_all":
+        help_str = "Required: (str, single or multiple) ('all' for all)"
+
+    return help_str
+
+
+# ------------------------------------------------------------------------------------------
+# Entry
+# ------------------------------------------------------------------------------------------
+
+description = (
+    "----------------------------------------------------------------------\n"
+    "DataShuttle command line interface. "
+    "\n\n"
+    "datashuttle [PROJECT NAME] [COMMAND] [OPTIONS]"
+    "\n\n"
+    "To get detailed help for commands and their optional arguments, "
+    "\ntype python -m datashuttle [PROJECT NAME] [COMMAND] --help'"
+    "\n\n"
+    "On first use it is necessary to setup configurations. \ne.g."
+    "'python -m datashuttle [PROJECT NAME] make_config_file [OPTIONS]'"
+    "\n\n"
+    "see \n'python -m datashuttle <project_name> make_config_file --help'"
+    "\nfor full list of options."
+    "\n\n"
+    "All command and argument names are matched to the API "
+    "documentation. "
+    "\n\n"
+    "----------------------------------------------------------------------"
+)
+
+parser = argparse.ArgumentParser(
+    prog="datashuttle",
+    usage="%(prog)s [PROJECT NAME]",
```

Comment:
I am curious. Is there a specific reason to use the old printing style instead of f"{prog} [PROJECT NAME]"?

Reply:
good question! it was just how I copied it from SO, but just tested it and new the new way does not work (as progs is not defined in this scope and I guess is handled internally from the string by argparse). Have added a comment!